### ISSUE TITLE
Adding tag value for accuracy

### ISSUE THREAD

**Issue by tarbault** [nemethviktor/GeoTagNinja#121]
**Is your feature request related to a problem? Please describe.**
I use GeotagNinja to tag images of plants and animals to upload them on iNaturalist.
iNaturalist "requires" if possible the accuracy to be present to assess the quality of an observation.
Most research studies discard observations missing accuracy.

**Describe the solution you'd like**
use the value of the HDOP present in the GPX to fill the tag named GPSHPositioningError 

sample tag is: `<hdop>1.3</hdop>`
Sample gpx file attache with a full line

**Describe alternatives you've considered**
I found no software that can do that as of today

**Additional context**
[20241102.txt](https://github.com/user-attachments/files/17611491/20241102.txt)

**Comment by nemethviktor** [comment]
Hi, 

I'll need a bit more context please as admittedly I don't understand what you would like at this stage. Also I'm not entirely sure that if I do understand correctly what you're after, it's possible. 

As for the request:
In the Import GPX (?? -- if not, shout pls.) you'd like to import the HDOP values (associated with a timestamp?) and then write those values as a new tag (?) called GPSHPositioningError

I've taken an arbitrary jpg file and aligned it with your txt file and did: `exiftool -geotag 20241102.txt -v4 20241027_Virginia_Water_EOS_50D_IMG_3383_MOD_20241028.JPG`
Where my output was then:
```
d:\temp>exiftool -geotag 20241102.txt -v4 20241027_Virginia_Water_EOS_50D_IMG_3383_MOD_20241028.JPG
Argument "-Geotime<DateTimeOriginal#" is assumed
Loaded 1 points from XML-format GPS track log file '20241102.txt'
  GPS track start: 2024:11:01 23:00:14.000 UTC
    2024:11:01 23:00:14.000 UTC - alt=74.24 first=1 hdop=2.9 lat=50.67 lon=3.15 nsats=13 pdop=4.3 speed=0.07628542 vdop=3.2
  GPS track end:   2024:11:01 23:00:14.000 UTC
Writing File:Geotag
======== 20241027_Virginia_Water_EOS_50D_IMG_3383_MOD_20241028.JPG
Setting new values from 20241027_Virginia_Water_EOS_50D_IMG_3383_MOD_20241028.JPG
  Geotime value:   2024:11:01 23:00:58.000 UTC (local timezone is +00:00)
  Taking pos from fix:
    2024:11:01 23:00:14.000 UTC - alt=74.24 first=1 hdop=2.9 lat=50.67 lon=3.15 nsats=13 pdop=4.3 speed=0.07628542 vdop=3.2
Writing Keys:GPSCoordinates if tag exists
Writing UserData:GPSCoordinates if tag exists
Writing UserData:GPSCoordinates if tag exists
Writing ItemList:GPSCoordinates
Writing MIE-GPS:GPSLatitude
Writing XMP-drone-dji:GPSLatitude if tag exists
Writing XMP-exif:GPSLatitude if tag exists
Writing GPS:GPSLatitude
Writing MIE-GPS:GPSLongitude
Writing XMP-drone-dji:GPSLongitude if tag exists
Writing XMP-exif:GPSLongitude if tag exists
Writing GPS:GPSLongitude
Writing MIE-GPS:GPSAltitude
Writing XMP-exif:GPSAltitude if tag exists
Writing GPS:GPSAltitude
Writing XMP-exif:GPSAltitudeRef if tag exists
Writing GPS:GPSAltitudeRef
Deleting MIE-GPS:GPSTrack
Deleting XMP-exif:GPSTrack
Deleting GPS:GPSTrack
Deleting XMP-exif:GPSTrackRef
Deleting GPS:GPSTrackRef
Writing MIE-GPS:GPSSpeed
Writing XMP-exif:GPSSpeed if tag exists
Writing GPS:GPSSpeed
Writing XMP-exif:GPSSpeedRef if tag exists
Writing GPS:GPSSpeedRef
Writing GPS:GPSLatitudeRef
Writing GPS:GPSLongitudeRef
Writing GPS:GPSDateStamp
Writing GPS:GPSTimeStamp
Writing XMP-exif:GPSDateTime if tag exists
Rewriting 20241027_Virginia_Water_EOS_50D_IMG_3383_MOD_20241028.JPG...
  Editing tags in: APP0 APP1 File GPS IFD0 ItemList JFIF Keys MIE-GPS UserData XMP
  Creating tags in: APP1 File GPS IFD0 ItemList MIE-GPS
JPEG APP1 (12736 bytes)
  Rewriting IFD0
  Rewriting ExifIFD
    [nothing changed in ExifIFD]
  Creating GPS
    + GPS:GPSVersionID = '2 3 0 0' (mandatory)
    + GPS:GPSLatitudeRef = 'N'
    + GPS:GPSLatitude = '50 40 12.0000000000063'
    + GPS:GPSLongitudeRef = 'E'
    + GPS:GPSLongitude = '3 8 59.9999999999997'
    + GPS:GPSAltitudeRef = '0'
    + GPS:GPSAltitude = '74.24'
    + GPS:GPSTimeStamp = '23 00 58'
    + GPS:GPSSpeedRef = 'N'
    + GPS:GPSSpeed = '0.07628542'
    + GPS:GPSDateStamp = '2024:11:01'
  Rewriting IFD1
JPEG APP13 (12090 bytes)
JPEG APP2 (574 bytes)
JPEG APP1 (14473 bytes)
  Rewriting XMP
    [nothing changed in XMP]
    [XMP rewritten with no changes]
JPEG DQT (130 bytes)
JPEG DRI (2 bytes)
JPEG APP14 (12 bytes)
JPEG SOF0:
JPEG DHT (184 bytes)
JPEG SOS
    1 image files updated

```

I guess what I need to understand is, where would I find the tag you're after? Exiftool appears to take notice of it but I can't seem to find it in the output. (`exiftool -a -G -u   20241027_Virginia_Water_EOS_50D_IMG_3383_MOD_20241028.JPG`)
![image](https://github.com/user-attachments/assets/70683881-d151-467a-8f5e-5393405a64a6)

At the same time doing some quick search Phil has suggested very recently that such transformations aren't possible: https://exiftool.org/forum/index.php?topic=16545.0
... so I'm a bit unsure. But in the first instance we'd need to be able to read and identify that particular tag.

**Comment by StarGeekSpaceNerd** [comment]
This question led me down a GPS rabbit hole. I believe technically, `hdop` or `pdop` should be saved to the `GPSDop` tag in EXIF, and I'm going to post this as a suggestion on the exiftool forums.

But it probably doesn't belong in `GPSHPositioningError`. I'm going by ChatGPT, which of course, should be double-checked. But it says that DoP values are not measurements, such as meters. It's a multiplier against the baseline accuracy. `GPSHPositioningError`  instead "indicates horizontal positioning errors in meters" according to the EXIF spec.

For what it's worth, [here's the ChatGPT](https://chatgpt.com/share/6727d1e6-7c70-800a-ad07-7b2c1dc3eb60) session.

**Comment by nemethviktor** [comment]
As always, thanks for the valuable input;)

So yes I think it's currently not something I can do but if Phil implements it in ET I can then transfer that into GTN

**Comment by StarGeekSpaceNerd** [comment]
Digging further down the rabbit hole, it looks like the only time exiftool will geotag `GPSHPositioningError` is when it encounters `accuracy_horizontal` which appears in [opentrack GPX files](https://github.com/opentrack/opentrack). [Source](https://exiftool.org/forum/index.php?msg=85869) of that addition.

**Comment by tarbault** [comment]
Hello,

Thanks to both of you for the quick answer and the time you took to do it.

First, yes, I'm also Mobilis, my mistake for not checking which account was
active at the time, and... well done Sherlock !😉

For some more context (that I realise I could have provided to start with,
sorry), iNaturalist is both a website and database to provide "citizen
scientists" with a way to report the presence of animal and plant species
they observe around them.
Observations are qualified, needing a recording (picture or sound, mostly
the former), an identification, a date, time and position, preferably using
a GPS, and if possible mentioning accuracy.

*Using the App*
More to the point, the iNaturalist organisation also provides an app for
smartphones that uses the phone's camera and gps to create records and
upload them.
It also provides "computer vision" that uses AI to identify what the camera
sees in real time.
When using the iNat app internal camera, it calculates and stores a value
for accuracy in meters, which is the behaviour I'm trying to replicate when
publishing from the computer (see below)
It is not the case if the picture is taken from the camera software outside
the app, and then published later through the iNat app using the upload
from gallery option, so this calculation is made by the app itself and not
by the smartphone camera software.

*Using a computer*
A lot of users publish their observations from a computer via the website,
which I prefer, smartphone pictures not being good enough for some
situations (macro photography, birds etc.)

This is where my question comes in: is there a way to write a value for
accuracy in the tags of the picture before uploading it ?
Some research on the iNat forum indicates that it is possible:

> (e.g. the “GPSHPositioningError” EXIF tag, which is successfully
> interpreted as ‘accuracy’ by iNat, at the moment).

So I went looking for a way to do this, and... here we are !

I also noted that some cameras do store more GPS data than others (basics
are Lat, Lon, Alt, of course, but some store speed, direction and...
accuracy), so I thought, "why not ask ?"

Of course, I do realise that this functionality is not something that will
matter to everyone, and that it currently might ot even be possible, period.
But it is my contribution to "put it on the radar" just in case...
Thanks again for everything !


*References*
I had done some research before posting my request, and you are absolutely
right: HDOP is not a measure in meters, but a value that is used to
calculate it.

Here is some reference to the subject:

> > Calculation Process
>
> To estimate GPS accuracy from HDOP, you need to combine it with another
> factor:
>
> UERE (User Equivalent Range Error): This represents the error in the
> measurement of the distance to each satellite.
>
> Calculation Formula:
> The formula to estimate horizontal position accuracy is: Estimated
> Accuracy=HDOP×UERE
>
> Steps to Calculate
>
> Determine HDOP: This value is typically provided by your GPS receiver.
>
> Estimate UERE: This value can vary depending on the GPS receiver quality
> and environmental conditions. A common estimate for civilian GPS is about 5
> meters.
>
> Apply the Formula: Multiply HDOP by UERE.
> >
> > For example:
> > If HDOP = 1.5 and UERE = 5 meters:
> >>
> >> Estimated Accuracy=1.5×5=7.5 meters


------------------------------
*Thierry Arbault - iNaturalist
<https://www.inaturalist.org/observations?user_id=thierrya>*


Le dim. 3 nov. 2024 à 21:19, StarGeekSpaceNerd ***@***.***> a
écrit :

**Comment by nemethviktor** [comment]
Hi, 

I think there are two separate issues here. 
The simple one, "is it possible to write that particular tag"? That is a yes and if you just do 

`exiftool -GPSHPositioningError=0.2 d:\temp\20241027_Virginia_Water_EOS_50D_IMG_3383_MOD_20241028.JPG`

You'll see it in the file:
![image](https://github.com/user-attachments/assets/070fdc52-3a30-4845-bd37-6f79d6fcd4cd)

The other is whether we can read that tag from your gpx/txt file and _then_ relay it further and at the moment I think that's not possible because ET doesn't read that parameter as part of the gpx import.

**Comment by tarbault** [comment]
That would be great, thanks!

------------------------------
*Thierry Arbault - iNaturalist
<https://www.inaturalist.org/observations?user_id=thierrya>*


Le lun. 4 nov. 2024 à 12:30, Viktor Nemeth ***@***.***> a
écrit :

**Comment by StarGeekSpaceNerd** [comment]
On the exiftool side, if the `GPSDop` tag is filled and assuming 5 meters for the baseline range error, the exiftool command would be

`exiftool "-GPSHPositioningError<${GPSDop;$_*=5}" /path/to/files/`

This command creates backup files.  Add [`-overwrite_original`](https://exiftool.org/exiftool_pod.html#overwrite_original) to suppress the creation of backup files.  Add [`-r`](https://ExifTool.org/exiftool_pod.html#r-.--recurse) to recurse into subdirectories.  If this command is run under Unix/Mac, reverse any double/single quotes to avoid bash interpretation.

**Comment by tarbault** [comment]
Thanks, I'll check it out.

------------------------------
*Thierry Arbault - iNaturalist
<https://www.inaturalist.org/observations?user_id=thierrya>*


Le lun. 4 nov. 2024 à 15:51, StarGeekSpaceNerd ***@***.***> a
écrit :

**Comment by tarbault** [comment]
That is excellent ! Now it's on the radar, and it might be possible in the
future.
Many thanks again

------------------------------
*Thierry Arbault - iNaturalist
<https://www.inaturalist.org/observations?user_id=thierrya>*


Le lun. 4 nov. 2024 à 19:13, StarGeekSpaceNerd ***@***.***> a
écrit :

**Comment by nemethviktor** [comment]
Right, anyway, Phil has patched the code in ET 13.04 for `GPSDOP` so now you have

```
Microsoft Windows [Version 10.0.22635.4440]
(c) Microsoft Corporation. All rights reserved.

d:\temp>exiftool -geotag 20241102.txt -v4 20241018_Foggy_Commute_IMG_3279_MOD_20241028.JPG
Argument "-Geotime<DateTimeOriginal#" is assumed
Loaded 1 points from XML-format GPS track log file '20241102.txt'
  GPS track start: 2024:11:01 23:00:14.000 UTC
    2024:11:01 23:00:14.000 UTC - alt=74.24 first=1 hdop=2.9 lat=50.67 lon=3.15 nsats=13 pdop=4.3 speed=0.07628542 vdop=3.2
  GPS track end:   2024:11:01 23:00:14.000 UTC
Writing File:Geotag
======== 20241018_Foggy_Commute_IMG_3279_MOD_20241028.JPG
Setting new values from 20241018_Foggy_Commute_IMG_3279_MOD_20241028.JPG
  Geotime value:   2024:11:01 23:00:14.000 UTC (local timezone is +00:00)
  Interpolating between fixes (f=0):
    2024:11:01 23:00:14.000 UTC - alt=74.24 first=1 hdop=2.9 lat=50.67 lon=3.15 nsats=13 pdop=4.3 speed=0.07628542 vdop=3.2
    2024:11:01 23:00:14.000 UTC - alt=74.24 first=1 hdop=2.9 lat=50.67 lon=3.15 nsats=13 pdop=4.3 speed=0.07628542 vdop=3.2
Writing Keys:GPSCoordinates if tag exists
Writing UserData:GPSCoordinates if tag exists
Writing UserData:GPSCoordinates if tag exists
Writing ItemList:GPSCoordinates
Writing MIE-GPS:GPSLatitude
Writing XMP-drone-dji:GPSLatitude if tag exists
Writing XMP-exif:GPSLatitude if tag exists
Writing GPS:GPSLatitude
Writing MIE-GPS:GPSLongitude
Writing XMP-drone-dji:GPSLongitude if tag exists
Writing XMP-exif:GPSLongitude if tag exists
Writing GPS:GPSLongitude
Writing MIE-GPS:GPSAltitude
Writing XMP-exif:GPSAltitude if tag exists
Writing GPS:GPSAltitude
Writing XMP-exif:GPSAltitudeRef if tag exists
Writing GPS:GPSAltitudeRef
Deleting MIE-GPS:GPSTrack
Deleting XMP-exif:GPSTrack
Deleting GPS:GPSTrack
Deleting XMP-exif:GPSTrackRef
Deleting GPS:GPSTrackRef
Deleting MIE-GPS:GPSSpeed
Deleting XMP-exif:GPSSpeed
Deleting GPS:GPSSpeed
Deleting XMP-exif:GPSSpeedRef
Deleting GPS:GPSSpeedRef
Writing MIE-GPS:GPSMeasureMode
Writing XMP-exif:GPSMeasureMode if tag exists
Writing GPS:GPSMeasureMode
Writing XMP-exif:GPSDOP if tag exists
Writing GPS:GPSDOP
Writing GPS:GPSLatitudeRef
Writing GPS:GPSLongitudeRef
Writing GPS:GPSDateStamp
Writing GPS:GPSTimeStamp
Writing XMP-exif:GPSDateTime if tag exists
  Geotime value:   2024:11:01 23:00:14.000 UTC (local timezone is +00:00)
  Interpolating between fixes (f=0):
    2024:11:01 23:00:14.000 UTC - alt=74.24 first=1 hdop=2.9 lat=50.67 lon=3.15 nsats=13 pdop=4.3 speed=0.07628542 vdop=3.2
    2024:11:01 23:00:14.000 UTC - alt=74.24 first=1 hdop=2.9 lat=50.67 lon=3.15 nsats=13 pdop=4.3 speed=0.07628542 vdop=3.2
Writing Keys:GPSCoordinates if tag exists
Writing UserData:GPSCoordinates if tag exists
Writing UserData:GPSCoordinates if tag exists
Writing ItemList:GPSCoordinates
Writing MIE-GPS:GPSLatitude
Writing XMP-drone-dji:GPSLatitude if tag exists
Writing XMP-exif:GPSLatitude if tag exists
Writing GPS:GPSLatitude
Writing MIE-GPS:GPSLongitude
Writing XMP-drone-dji:GPSLongitude if tag exists
Writing XMP-exif:GPSLongitude if tag exists
Writing GPS:GPSLongitude
Writing MIE-GPS:GPSAltitude
Writing XMP-exif:GPSAltitude if tag exists
Writing GPS:GPSAltitude
Writing XMP-exif:GPSAltitudeRef if tag exists
Writing GPS:GPSAltitudeRef
Deleting MIE-GPS:GPSTrack
Deleting XMP-exif:GPSTrack
Deleting GPS:GPSTrack
Deleting XMP-exif:GPSTrackRef
Deleting GPS:GPSTrackRef
Deleting MIE-GPS:GPSSpeed
Deleting XMP-exif:GPSSpeed
Deleting GPS:GPSSpeed
Deleting XMP-exif:GPSSpeedRef
Deleting GPS:GPSSpeedRef
Writing MIE-GPS:GPSMeasureMode
Writing XMP-exif:GPSMeasureMode if tag exists
Writing GPS:GPSMeasureMode
Writing XMP-exif:GPSDOP if tag exists
Writing GPS:GPSDOP
Writing GPS:GPSLatitudeRef
Writing GPS:GPSLongitudeRef
Writing GPS:GPSDateStamp
Writing GPS:GPSTimeStamp
Writing XMP-exif:GPSDateTime if tag exists
Rewriting 20241018_Foggy_Commute_IMG_3279_MOD_20241028.JPG...
  Editing tags in: APP0 APP1 File GPS IFD0 ItemList JFIF Keys MIE-GPS UserData XMP
  Creating tags in: APP1 File GPS IFD0 ItemList MIE-GPS
JPEG APP1 (13868 bytes)
  Rewriting IFD0
  Rewriting ExifIFD
    [nothing changed in ExifIFD]
  Creating GPS
    + GPS:GPSVersionID = '2 3 0 0' (mandatory)
    + GPS:GPSLatitudeRef = 'N'
    + GPS:GPSLatitude = '50 40 12.0000000000063'
    + GPS:GPSLongitudeRef = 'E'
    + GPS:GPSLongitude = '3 8 59.9999999999997'
    + GPS:GPSAltitudeRef = '0'
    + GPS:GPSAltitude = '74.24'
    + GPS:GPSTimeStamp = '23 00 14'
    + GPS:GPSMeasureMode = '3'
    + GPS:GPSDOP = '4.3'
    + GPS:GPSDateStamp = '2024:11:01'
  Rewriting IFD1
JPEG APP13 (13220 bytes)
JPEG APP2 (574 bytes)
JPEG APP1 (17262 bytes)
  Rewriting XMP
    [nothing changed in XMP]
    [XMP rewritten with no changes]
JPEG DQT (130 bytes)
JPEG DRI (2 bytes)
JPEG APP14 (12 bytes)
JPEG SOF0:
JPEG DHT (175 bytes)
JPEG SOS
    1 image files updated

d:\temp>
```
I'll update the GTN codebase to take GPSDOP. 
I'm not sure this is what you wanted as it's not precisely the tag you were after but if that's the case can you talk to him on the ET forum please as functionality need to be added in to ET so that I can amend it on my side.

**Comment by nemethviktor** [comment]
![image](https://github.com/user-attachments/assets/7002e663-d4a5-4001-92f7-f5e6e9b52110)

https://drive.google.com/open?id=13uYyCQE31JlSUb2iamKdYh55bGNqQFOW&usp=drive_fs
